BrowserKit/Sources/SiteImageView/URLProcessing/Cache/FaviconURLCache.swift

```diff
@@ -0,0 +1,85 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0
+
+import Foundation
+
+protocol FaviconURLCache {
+    func getURLFromCache(domain: String) async throws -> URL
+    func cacheURL(domain: String, faviconURL: URL) async
+}
+
+actor DefaultFaviconURLCache: FaviconURLCache {
+    private enum CacheConstants {
+        static let cacheKey = "favicon-url-cache"
+    }
+
+    static let shared = DefaultFaviconURLCache()
+    private let fileManager: URLCacheFileManager
+    private var urlCache = [String: FaviconURL]()
+    private var preserveTask: Task<Void, Never>?
+    private let preserveDebounceTime: UInt64 = 5_000_000_000 // 5 seconds
+
+    init(fileManager: URLCacheFileManager = DefaultURLCacheFileManager()) {
+        self.fileManager = fileManager
+
+        Task {
+            await retrieveCache()
+        }
+    }
+
+    func getURLFromCache(domain: String) async throws -> URL {
+        guard let favicon = urlCache[domain],
+              let url = URL(string: favicon.faviconURL) else {
+            throw SiteImageError.noURLInCache
+        }
```

Comment:
```suggestion
        guard let favicon = urlCache[domain],
              let url = URL(string: favicon.faviconURL)
        else { throw SiteImageError.noURLInCache }
```
nit: `else` on newline if we're doing multi-line guard statements. And as the else is one line of code, it can remain one line. Probably don't accept this suggestion as GH is not good with indenting though....